app/controllers/api/actions_controller.rb

```diff
@@ -27,6 +27,14 @@ def validate
     end
   end
 
+  def publish
+    action = Action.find(params[:id])
+    action.update(publish_status: params[:publish_status])
+    render json: {}, status: 200
+  rescue ArgumentError => e
+    render json: { errors: { publish_status: e.message } }, status: 422
+  end
```

Comment:
> Sending resource params at the root level is not a good practice, would you mind sending them instead as params[:action][:publish_status]?

I'm just using the same style as the existing code in the controller. I could update the rest of the params in the controller to have action as the root param, but that means we'd have to update the form liquid templates and invalidate the whole cache, which is more than I want to get into.
> Rescuing an ArgumentError is not very explicit or clear about what kind of error are you catching. Is there any reason why you aren't using AR validations?
Enums don't use AR validations properly - see [this issue on the Rails core repo](https://github.com/rails/rails/issues/13971).
> Calling update and not checking for the return value can cause the update to fail silently. It's recommended that you either check for the return value or use the bang version update!.
Because of the above, in this case `update` and `update!` are identical. But I'll change to `update!` for clarity.